From 31350bc0b935920f9924317b4cb3602602420f83 Mon Sep 17 00:00:00 2001
From: Jiri Vymazal <jvymazal@redhat.com>
Date: Fri, 16 Nov 2018 13:16:13 +0100
Subject: [PATCH] disable file vs directory error on symlinks

The file/directory node-object alignment now ignores symlinks.
Previously it reported error on each directory symlink spamming
user error logs.
---
 plugins/imfile/imfile.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/imfile/imfile.c b/plugins/imfile/imfile.c
index f42670ca2..1618d151c 100644
--- a/plugins/imfile/imfile.c
+++ b/plugins/imfile/imfile.c
@@ -869,7 +869,7 @@ poll_tree(fs_edge_t *const chld)
 					"directory - ignored", file);
 				continue;
 			}
-			if(chld->is_file != is_file) {
+			if(!issymlink && (chld->is_file != is_file)) {
 				LogMsg(0, RS_RET_ERR, LOG_WARNING,
 					"imfile: '%s' is %s but %s expected - ignored",
 					file, (is_file) ? "FILE" : "DIRECTORY",
